test/util_test.dart

```diff
@@ -125,6 +125,35 @@ void overReactExample() {}''';
       });
     });
 
+    group('generateNewVersionRange()', () {
+      group('updates correctly with a basic range', () {
+        sharedGenerateNewVersionRangeTests(
+          VersionConstraint.parse('>=0.5.0 <3.0.0'),
+          VersionConstraint.parse('>=1.5.0 <3.0.0'),
+          VersionConstraint.parse('>=1.0.0 <4.0.0'),
+          VersionConstraint.parse('>=1.5.0 <4.0.0'),
+        );
+      });
+
+      group('updates correctly with an open ended target range', () {
+        sharedGenerateNewVersionRangeTests(
+          VersionConstraint.parse('>=1.0.0 <2.0.0'),
+          VersionConstraint.parse('>=1.2.0 <2.0.0'),
+          VersionConstraint.parse('>=1.0.0'),
+          VersionConstraint.parse('>=1.2.0'),
+        );
+      });
+
+      group('updates correctly with an open ended current range', () {
+        sharedGenerateNewVersionRangeTests(
```

Comment:
These invocations would be a lot more readable if the arguments were named instead of positional